finality-aleph/src/justification/compatibility.rs

```diff
@@ -57,12 +57,17 @@ enum VersionedAlephJustification {
 }
 
 fn encode_with_version(version: Version, mut payload: Vec<u8>) -> Vec<u8> {
-    let mut result = version.encode();
+    let mut result =
+        Vec::with_capacity(size_of::<Version>() + size_of::<ByteCount>() + payload.len());
```

Comment:
`size_of` and size of encoded `Version` might be different. Use `size_hint` instead?